ios/app/brave_profile_controller.mm

```diff
@@ -0,0 +1,411 @@
+  if (!_braveWalletAPI) {
+    _braveWalletAPI = [[BraveWalletAPI alloc] initWithBrowserState:_profile];
+  }
+  return _braveWalletAPI;
+}
+
+- (DeAmpPrefs*)deAmpPrefs {
+  if (!_deAmpPrefs) {
+    _deAmpPrefs =
+        [[DeAmpPrefs alloc] initWithProfileState:_profile->GetPrefs()];
+  }
+  return _deAmpPrefs;
+}
+
+- (AIChat*)aiChatAPIWithDelegate:(id<AIChatDelegate>)delegate {
+  return [[AIChat alloc] initWithProfileIOS:_profile delegate:delegate];
+}
+
+- (DefaultHostContentSettings*)defaultHostContentSettings {
+  if (!_defaultHostContentSettings) {
+    HostContentSettingsMap* map =
+        ios::HostContentSettingsMapFactory::GetForProfile(_profile);
+    _defaultHostContentSettings =
+        [[DefaultHostContentSettings alloc] initWithSettingsMap:map];
+  }
+  return _defaultHostContentSettings;
+}
+
+- (CWVWebViewConfiguration*)defaultWebViewConfiguration {
+  if (!_defaultWebViewConfiguration) {
+    _defaultWebViewConfiguration = [[CWVWebViewConfiguration alloc]
+        initWithBrowserState:ios_web_view::WebViewBrowserState::
+                                 FromBrowserState(_profile)];
+  }
+  return _defaultWebViewConfiguration;
+}
+
+- (CWVWebViewConfiguration*)nonPersistentWebViewConfiguration {
+  if (!_nonPersistentWebViewConfiguration) {
+    _nonPersistentWebViewConfiguration = [[CWVWebViewConfiguration alloc]
+        initWithBrowserState:ios_web_view::WebViewBrowserState::
+                                 FromBrowserState(
+                                     _profile->GetOffTheRecordProfile())];
+  }
+  return _nonPersistentWebViewConfiguration;
+}
+
+#pragma mark - Handling of destroying the incognito BrowserState
+
+// The incognito BrowserState should be closed when the last incognito tab is
+// closed (i.e. if there are other incognito tabs open in another Scene, the
+// BrowserState must not be destroyed).
+- (BOOL)shouldDestroyAndRebuildIncognitoProfile {
+  return _profile->HasOffTheRecordProfile();
+}
+
+// Matches lastIncognitoTabClosed from Chrome's SceneController
+- (void)notifyLastPrivateTabClosed {
+  // If no other window has incognito tab, then destroy and rebuild the
+  // BrowserState. Otherwise, just do the state transition animation.
+  if ([self shouldDestroyAndRebuildIncognitoProfile]) {
+    // Incognito browser state cannot be deleted before all the requests are
+    // deleted. Queue empty task on IO thread and destroy the BrowserState
+    // when the task has executed, again verifying that no incognito tabs are
+    // present. When an incognito tab is moved between browsers, there is
+    // a point where the tab isn't attached to any web state list. However, when
+    // this queued cleanup step executes, the moved tab will be attached, so
+    // the cleanup shouldn't proceed.
+
+    auto cleanup = ^{
+      if ([self shouldDestroyAndRebuildIncognitoProfile]) {
+        [self destroyAndRebuildIncognitoProfile];
+      }
+    };
+
+    web::GetIOThreadTaskRunner({})->PostTaskAndReply(
```

Comment:
nit: AndReply not needed? The function doesn't return anything, so I think PostTask without the reply would work.